### PR TITLE
 Coerce fails with string colspec

### PR DESCRIPTION
**Pull request by Jay-sanjay**
Coerce fails with string colspec   issue #194  fixing
  -->  Added extra line in coerce.jl for string implementation

### CODE REVIEW

**Review by juliohm — CHANGES_REQUESTED** [review]
Thank you @Jay-sanjay for the proposed fix. Can you please add tests as well?

**Comment by Jay-sanjay** [comment]
yes sure , I will work on that

File: src/transforms/coerce.jl [hunk 1]
```diff
@@ -23,6 +23,7 @@ struct Coerce{P} <: FeatureTransform
 end
 
 Coerce(pair::Pair{Symbol,<:Type}...; tight=false, verbosity=1) = Coerce(pair, tight, verbosity)
+Coerce(pair::Pair{String,<:Type}...; tight=false, verbosity=1) = Coerce(Symbol(pair)..., tight, verbosity)
```
Comment (eliascarv) [inline src/transforms/coerce.jl:26]: I don't think this code will work, try this one:
```suggestion
Coerce(pair::Pair{<:AbstractString,<:Type}...; kwargs...) = Coerce((Symbol(k) => v for (k, v) in pairs)...; kwargs....)
```

File: src/transforms/coerce.jl [hunk 2]
```diff
@@ -23,6 +23,7 @@ struct Coerce{P} <: FeatureTransform
 end
 
 Coerce(pair::Pair{Symbol,<:Type}...; tight=false, verbosity=1) = Coerce(pair, tight, verbosity)
+Coerce(pair::Pair{<:AbstractString,<:Type}...; kwargs...) = Coerce((Symbol(k) => v for (k, v) in pairs)...; kwargs...)
```
Comment (eliascarv) [inline src/transforms/coerce.jl:26]:
```suggestion
Coerce(pairs::Pair{Symbol,<:Type}...; tight=false, verbosity=1) = Coerce(pairs, tight, verbosity)
Coerce(pairs::Pair{<:AbstractString,<:Type}...; kwargs...) = Coerce((Symbol(k) => v for (k, v) in pairs)...; kwargs...)
```

File: test/transforms/coerce.jl [hunk 3]
```diff
@@ -22,6 +23,16 @@
   @test eltype(tₒ.x1) == eltype(t.x1)
   @test eltype(tₒ.x2) == eltype(t.x2)
 
+  # Test for "a" => Multiclass
```
Comment (eliascarv) [inline test/transforms/coerce.jl:26]: I think comments are not necessary in this case.

**Comment by juliohm** [comment]
Feel free to merge.

Em ter., 5 de set. de 2023 11:53, Elias Carvalho ***@***.***>
escreveu:

**Comment by juliohm** [comment]
Thank you @Jay-sanjay ! Feel free to pick another one. We can guide you with tips and suggestions.

**Pull request merged**